src/core/tsi/ssl_transport_security.cc

```diff
@@ -439,7 +463,7 @@ static tsi_result peer_from_x509(X509* cert, int include_certificate_type,
   tsi_result result;
   GPR_ASSERT(subject_alt_name_count >= 0);
   property_count = (include_certificate_type ? static_cast<size_t>(1) : 0) +
-                   2 /* common name, certificate */ +
+                   3 /* subject, common name, certificate */ +
```

Comment:
Meh...we really should improve the code here to be more C++, and avoid things like this...can we at least have a bug to track this effort?
CC @yihuazhang for some potential work